File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 8/16/2026, 9:25:28 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
     
        
     unordered_map<int , int > m;
     for(int i = 0 ; i < nums.size() ;i++){
        int com = target - nums[i];
        if(m.find(com) != m.end()){
            return {m[com] , i};
        }
        m[nums[i]] = i;
     } 
    return { };
    }
};

File: 1-TwoSum/1-TwoSum.js
// Last updated: 8/16/2026, 9:25:30 PM
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  var arr = [] 
  for(let i = 0 ; i < nums.length ; i++){
for(let j = i+1 ; j < nums.length ; j++){
if(nums[i]+nums[j] == target){
arr[0] = i;
arr[1] = j;
}
}
  }
return arr;
};
